components/MapComponent.tsx

```diff
@@ -1,170 +1,127 @@
+  const [currentMarkers, setCurrentMarkers] = useState(markers);
+
+  /*const onLongPress = () => {
+    setCurrentMarkers([...currentMarkers, e.nativeEvent.coordinate]);
+  }*/
+  
+  const mapMarkerCreations = () => {
+  return markers.map((report) =>
+    <Marker
+    key = {report.key}
+    title = {report.place}
+    coordinate = {{ latitude: report.lat, longitude: report.lng }}
+    pinColor = {report.pinColor}
+    onPress={() => {onMarkerClick(report.lat, report.lng); setButtonVisible(true); setSelectedMarker(report)}}
+    icon={require('../assets/icon.png')}
+    >
+    </Marker >)
+  }
 
-// IDK PUT THE LOCATION YOU FETCH FROM THE LOCATION SERVICES HERE? Might be different, could just be placeholder
-const area = {
-  latitude: 34.059761,
-  longitude: -118.276802,
-  latitudeDelta: 0.0922,
-  longitudeDelta: 0.0421
-}
+  function onMarkerClick (lat: number, long: number) {
+    mapRef.current.animateToRegion({
+      latitude: lat,
+      longitude: long,
+      latitudeDelta: initialRegion.latitudeDelta - 0.02,
+      longitudeDelta: initialRegion.longitudeDelta - 0.02,
+    })
+    
+  }
 
-// Function to use map in order to go through objects and use information from them, for marker locations/information
-const mapMarkerCreations = () => {
-  return state.markers.map((report) =>
-  <Marker
-  key = {report.key}
-  title = {report.place}
-  coordinate = {{ latitude: report.lat, longitude: report.lng }}
-  >
-  </Marker >)
-}
+  function onRecenter () {
+    mapRef.current.animateToRegion({
+      latitude: initialRegion.latitude,
+      longitude: initialRegion.longitude,
+      latitudeDelta: initialRegion.latitudeDelta,
+      longitudeDelta: initialRegion.longitudeDelta,
+    })
+  }
 
-// Map view component itself
-function MapComponent () {
   return (
-    <View style={styles.container}>
+    <View style={containStyle}>
       <MapView
-          style={styles.map}
+          style={mapStyle}
           provider={PROVIDER_GOOGLE}
-          region={area}
-          rotateEnabled={false}
-          scrollEnabled={false}
+          initialRegion={initialRegion}
+          ref={mapRef}
+          rotateEnabled={movementSettings.canRotate}
+          scrollEnabled={movementSettings.canScroll}
+          followsUserLocation={true}
+          showsUserLocation={true}
+          zoomEnabled={movementSettings.canZoom}
+          customMapStyle={[
+            {
+              "featureType": "poi",
+              "stylers": [{ "visibility": "off" }]
+            },{
+              "featureType": "transit",
+              "stylers": [{ "visibility": "off" }]
+            }
+          ]}
       >
       {mapMarkerCreations()}
       </MapView>
+      <View
+        style={{
+        position: 'absolute',//use absolute position to show button on top of the map
+        top: '70%', //for center align
+        alignSelf: 'flex-end' //for align to right
+        }}
+      >
+      { 
+        buttonVisible == true &&
```

Comment:
`== true` shouldn't be needed when doing this comparison. If you need to check for if the value is simply not undefined you can do `!!buttonVisible && <Component />`.